msdbook/tests/test_fishery_dynamics.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.testing.decorators import check_figures_equal
+from fishery_dynamics import plot_objective_performance, plot_factor_performance
+
+@pytest.fixture
+def sample_data():
+    """Fixture to provide sample data for testing."""
+    objective_performance = np.array([
+        [1, 2, 3, 4, 5],
+        [2, 3, 4, 5, 6],
+        [3, 4, 5, 6, 7]
+    ])
+    profit_solution = 1
+    robust_solution = 2
+    param_values = np.random.rand(100, 7)  # Example data
+    collapse_days = np.random.rand(100, 2) * 10  # Example data
+    b = np.linspace(0, 1, 10)
+    m = np.linspace(0, 1, 10)
+    a = np.linspace(0, 2, 10)
+    return {
+        'objective_performance': objective_performance,
+        'profit_solution': profit_solution,
+        'robust_solution': robust_solution,
+        'param_values': param_values,
+        'collapse_days': collapse_days,
+        'b': b,
+        'm': m,
+        'a': a
+    }
+
+def test_plot_objective_performance(sample_data, mocker):
```

Comment:
you should add `pytest-mock` as a test dependency in `pyproject.toml`

Reply:
Is the pyproject.tmol? https://github.com/IMMM-SFA/naturf/blob/main/pyproject.toml